pkg/mutation/schema/node_test.go

```diff
@@ -154,6 +167,47 @@ func TestNode_Add(t *testing.T) {
 				id("list-object"):   true,
 			},
 		},
+		{
+			name: "don't need to terminate",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", false),
+			},
+			add:  ip("more fields", "spec.fields.foo.bar"),
+			want: nil,
+		},
+		{
+			name: "must terminate for before",
+			before: []idPath{
+				ipmt("object", "spec.fields.foo", true),
+			},
+			add: ip("more fields", "spec.fields.foo.bar"),
+			want: IDSet{
+				id("object"):      true,
+				id("more fields"): true,
+			},
+		},
+		{
+			name: "must terminate for after",
+			before: []idPath{
+				ip("more fields", "spec.fields.foo.bar"),
+			},
+			add: ipmt("object", "spec.fields.foo", true),
+			want: IDSet{
+				id("object"):      true,
+				id("more fields"): true,
+			},
+		},
+		{
+			name: "must terminate for before and after",
```

Comment:
For sanity, a pre-existing "must terminate" that  doesn't cause a conflict with the added constraint? e.g. spec.foo.bar.baz must terminate, but we are adding a mutator for spec.foo.bar

Reply:
And maybe at the same depth as well